vectordraw.js

```diff
@@ -222,16 +264,31 @@ VectorDraw.prototype.removeVector = function(idx) {
     if (object) {
         this.board.removeAncestors(object);
         // Enable the <option> element corresponding to vector.
-        var option = this.element.find('.menu option[value=' + idx + ']');
+        var option = this.element.find('.menu option[value=vector-' + idx + ']');
         option.prop('disabled', false);
     }
 };
 
-VectorDraw.prototype.addVectorFromList = function() {
+VectorDraw.prototype.getSelectedElement = function() {
+    var selector = this.element.find('.menu select').val();
+    if (selector) {
+        selector = selector.split('-');
```

Comment:
@smarnach nit: looks a bit like a string typization. I believe using data attributes would make it cleaner.